openedx/core/djangoapps/content/course_overviews/management/commands/backfill_history.py

```diff
@@ -0,0 +1,138 @@
+                columns = [column[0] for column in cursor.fetchall()]
+            if exclude_column in columns:
+                columns.remove(exclude_column)
+            columns = ['`{}`'.format(c) for c in columns]
+
+            with open(file_path) as input_file:
+                # Skip the header
+                input_file.readline()
+                while True:
+                    lines = input_file.readlines(byte_size)
+                    if not lines:
+                        break
+
+                    values = [line.strip('\n').split(',') for line in lines]
+                    # Add history columns data
+                    [value.extend([self.DATE, self.HISTORY_CHANGE_REASON, '+', self.HISTORY_USER_ID]) for value in values]
+                    # Convert to tuple
+                    values = [tuple(value) for value in values]
+                    # Get the ids
+                    ids = [value[0] for value in values]
+
+                    # Checks for existing historical records
+                    with connection.cursor() as cursor:
+                        query = u"""
+                            SELECT COUNT(1)
+                            FROM {historical_table}
+                            WHERE ID in ({ids})
+                            AND history_type='+'
+                            """.format(
+                                historical_table=historical_table,
+                                ids=','.join(ids)
+                            )
+                        log.info(query)
+                        cursor.execute(query)
+                        count = cursor.fetchone()[0]
+
+                    if count==len(ids):
+                        log.info(u"Initial history records already exist for ids %s..%s - skipping.", ','.join(ids[:2]), ','.join(ids[-2:]))
+                        continue
+                    elif count!=0:
+                        raise Exception(u"Database count: %s does not match input count: %s" % (count,len(ids)))
+
+                    with transaction.atomic():
+                        with connection.cursor() as cursor:
+                            query = u"""
+                                INSERT INTO {historical_table}(
+                                    {insert_columns},history_date,history_change_reason,history_type,history_user_id
+                                )
+                                VALUES ({placeholder})
+                                """.format(
+                                    historical_table=historical_table,
+                                    insert_columns=','.join(columns),
+                                    placeholder=','.join(['%s']*(len(columns)+4))
+                                )
+                            log.info(query)
+                            cursor.executemany(query, values)
+
+                    log.info("Sleeping %s seconds...", sleep_between)
+                    time.sleep(sleep_between)
```

Comment:
It would be nice to have the ability to restart from an offset in case only a few rows fail for whatever reason. If we could just print the current row at the end of the loop we could use that to restart from where we left off.